packages/mongodb-memory-server-core/src/util/__tests__/MongoInstance.test.ts

```diff
@@ -292,12 +292,30 @@ describe('MongodbInstance', () => {
     await mongod.start();
     jest.spyOn(MongoClient, 'connect');
     process.kill(mongod.mongodProcess!.pid, 'SIGKILL');
-    await new Promise<void>((resolve) => setTimeout(resolve, 100));
+    while (dbUtil.isAlive(mongod.mongodProcess!.pid)) {
+      await new Promise<void>((resolve) => setTimeout(resolve, 5));
+    }
     await mongod.stop();
 
     expect(MongoClient.connect).not.toBeCalled();
   });
 
+  test('"kill" should not wait too much to open a connection to ReplSet', async () => {
```

Comment:
```suggestion
  test('"kill" should not wait too much to open a connection', async () => {
```

from what i can tell, this is not replset specific
(the other PR also added a test with this in the title, could you remove it?)

Reply:
There's a condition `if (this.isReplSet)` to call `MongoClient.connect`. That's why I've written it in this way. I don't really know why its's done this way.

Reply:
> There's a condition if (this.isReplSet) to call MongoClient.connect
i am sorry, you are correct. i will re-review this again tomorrow when i am more awake again.

> I don't really know why its's done this way.

if i remember correctly, this is because if mongod is running as a replset and receives a signal to stop (like SIGTERM), it will have a ~30 second timeout before actually shutting down (trying to reach another server to synchronize to), but this does not happen if a `shutdown` with option `force: 1` command is given